src/main/java/com/minecolonies/coremod/inventory/InventoryCitizen.java

```diff
@@ -1034,13 +1034,13 @@ public void setItemStack(ItemStack itemStackIn)
      * @param itemStackIn the stack to be searched for.
      * @return true if it exists.
      */
-    public boolean hasItemStack(ItemStack itemStackIn)
+    public boolean hasItemStack(final ItemStack itemStackIn)
     {
         label19:
 
-        for (List<ItemStack> list : this.allInventories)
+        for (final List<ItemStack> list : this.allInventories)
         {
-            Iterator<ItemStack> iterator = list.iterator();
+            final Iterator<ItemStack> iterator = list.iterator();
 
             while (true)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)